gear-programs/erc20-relay/app/tests/gclient.rs

```diff
@@ -5,35 +5,68 @@ mod vft {
 }
 
 use erc20_relay_client::traits::{Erc20Relay, Erc20RelayFactory};
-use gclient::{Event, EventProcessor, GearApi, GearEvent};
+use gclient::{Event, EventProcessor, GearApi, GearEvent, WSAddress};
 use sails_rs::{calls::*, gclient::calls::*, prelude::*};
+use sp_core::crypto::DEV_PHRASE;
+use tokio::sync::Mutex;
 use vft::vft_manager;
 
-async fn spin_up_node() -> (GClientRemoting, GearApi, CodeId, GasUnit) {
+static LOCK: Mutex<(u32, Option<CodeId>)> = Mutex::const_new((0, None));
+
+async fn spin_up_node() -> (impl Remoting + Clone, GearApi, CodeId, GasUnit, [u8; 4]) {
     let api = GearApi::dev().await.unwrap();
     let gas_limit = api.block_gas_limit().unwrap();
-    let remoting = GClientRemoting::new(api.clone());
-    let (code_id, _) = api.upload_code(erc20_relay::WASM_BINARY).await.unwrap();
+    let (api, code_id, salt) = {
+        let mut lock = LOCK.lock().await;
+        let code_id = match lock.1 {
+            Some(code_id) => code_id,
+            None => {
+                let (code_id, _) = api.upload_code(erc20_relay::WASM_BINARY).await.unwrap();
+                lock.1 = Some(code_id);
+
+                code_id
+            }
+        };
+
+        let salt = lock.0;
+        lock.0 += 1;
+
+        let suri = format!("{DEV_PHRASE}//erc20-relay-{salt}:");
+        let api2 = GearApi::init_with(WSAddress::dev(), suri).await.unwrap();
+
+        let account_id: &[u8; 32] = api2.account_id().as_ref();
+        api.transfer_keep_alive((*account_id).into(), 100_000_000_000_000)
+            .await
+            .unwrap();
+
+        (api2, code_id, salt)
+    };
 
-    (remoting, api, code_id, gas_limit)
+    (
+        GClientRemoting::new(api.clone()),
+        api,
+        code_id,
+        gas_limit,
+        salt.to_le_bytes(),
+    )
 }
 
+#[ignore]
```

Comment:
This test is used for calculating minimal gas required to send `MintTokens` request so shouldn't be run every time. Moreover it will be removed in the PR with historical proxy implementation